From ee13ed2ae7bd074eb0a2923c3a7218019f8a3a35 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Wed, 19 Mar 2025 10:19:27 +0100
Subject: [PATCH 822/841] feat(stm32mp2): add IPCC base addresses

Add IPCC base addresses for STM32MP2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 57339c4af001f3acec8b048552c69e3e77c49876 feat(stm32mp2): add IPCC base addresses]
Change-Id: If87d08d1cc00451a1f7be5c6a96fbce6be966628
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/445180
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 8b25732ca..95889b20f 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -457,6 +457,14 @@ enum ddr_type {
 #define DEBUG_UART_PREDIV_CFGR			RCC_PREDIV8CFGR
 #define DEBUG_UART_FINDIV_CFGR			RCC_FINDIV8CFGR
 
+/*******************************************************************************
+ * STM32MP2 IPCC
+ ******************************************************************************/
+#define IPCC1_BASE				U(0x40490000)
+#if !STM32MP21
+#define IPCC2_BASE				U(0x46250000)
+#endif /* !STM32MP21 */
+
 /*******************************************************************************
  * STM32MP2 SDMMC
  ******************************************************************************/
-- 
2.47.2

